Add support for azure maps account

reshmarajan1730 opened the pull request:
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/1713)

## PR Checklist

---

<!-- Use the check list below to ensure your branch is ready for PR. -->

- [x] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

<!-- Concise description of the problem and the solution or the feature being added -->

## Does this introduce a breaking change

- [ ] YES
- [x] NO

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Testing
Tested standalone 
<!-- Instructions for testing and validation of your code -->

reshmarajan1730 commented:
@LaurentLesle renamed to maps_accounts

reshmarajan1730 commented:
@arnaudlh could you please have a look

arnaudlh commented on modules/maps/maps_account/maps_account.tf:
```diff
@@ -0,0 +1,15 @@
+# naming convention
+resource "azurecaf_name" "map" {
```
```suggestion
data "azurecaf_name" "map" {
```

reshmarajan1730 replied:
updated

arnaudlh commented on modules/maps/maps_account/maps_account.tf:
```diff
@@ -0,0 +1,15 @@
+  use_slug      = var.global_settings.use_slug
+}
+resource "azurerm_maps_account" "map" {
+  name                = azurecaf_name.map.result
```
```suggestion
  name                = data.azurecaf_name.map.result
```

reshmarajan1730 replied:
updated

arnaudlh commented on examples/variables.tf:
```diff
@@ -232,8 +232,8 @@ variable "mssql_mi_secondary_tdes" {
 variable "storage_accounts" {
   default = {}
 }
-variable "storage_account_file_shares" {
```
Do not delete other variables, just add yours :) 

```
variable "storage_account_file_shares" {
  default     = {}
}

variable "maps_accounts" {
  default     = {}
}
```

reshmarajan1730 replied:
updated

arnaudlh reviewed (changes requested):
thanks for your PR @reshmarajan1730! 

please have a look at my suggestions and comments and also please add support for tags as it is support by the maps per: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/maps_account#tags

reshmarajan1730 commented:
@arnaudlh I have updated the PR. With this update I have also added the feature to store the azure map account's primary and secondary access keys in a keyvault, if specified. This has also been tested.

arnaudlh reviewed (changes requested):
thanks @reshmarajan1730, its looks nice and works on my environment. 

Are you able to:
1. run a ```terraform fmt``` on your files 
2. add path to your example in  ```standalone-scenarios.json``` so the CD/CI picks it. 

After that we should be ready for merge! 
Thanks.

arnaudlh reviewed (approved):
LGTM

reshmarajan1730 commented:
@arnaudlh I am not able to merge because of the branch protection

arnaudlh commented:
> @arnaudlh I am not able to merge because of the branch protection

no worries, I'm the one merging ;)
